libraries/integration/Microsoft.Bot.Builder.Integration.AspNet.Core/BotMessageHandler.cs

```diff
@@ -16,12 +16,21 @@ public class BotMessageHandler : BotMessageHandlerBase
     {
         protected override async Task<InvokeResponse> ProcessMessageRequestAsync(HttpRequest request, IAdapterIntegration adapter, BotCallbackHandler botCallbackHandler, CancellationToken cancellationToken)
         {
+            var requestBody = request.Body;
+
+            // In the case of request buffering being enabled, we could possibly receive a Stream that needs its position reset,
+            // but we can't _blindly_ reset in case buffering hasn't been enabled since we'll be working with a non-seekable Stream
+            // in that case which will throw a NotSupportedException
+            if (requestBody.CanSeek)
```

Comment:
Only attempt to reposition the `Stream` when it's "seekable". `CanSeek` will be `true` in the buffered scenario and `false` in the non-buffered (default) scenario.